framework/src/main/java/org/checkerframework/common/accumulation/AccumulationTransfer.java

```diff
@@ -0,0 +1,151 @@
+package org.checkerframework.common.accumulation;
+
+import com.sun.source.tree.MethodInvocationTree;
+import com.sun.source.tree.Tree;
+import java.util.List;
+import javax.lang.model.element.AnnotationMirror;
+import org.checkerframework.common.value.ValueCheckerUtils;
+import org.checkerframework.dataflow.analysis.FlowExpressions;
+import org.checkerframework.dataflow.analysis.FlowExpressions.Receiver;
+import org.checkerframework.dataflow.analysis.TransferResult;
+import org.checkerframework.dataflow.cfg.node.MethodInvocationNode;
+import org.checkerframework.dataflow.cfg.node.Node;
+import org.checkerframework.framework.flow.CFAnalysis;
+import org.checkerframework.framework.flow.CFStore;
+import org.checkerframework.framework.flow.CFTransfer;
+import org.checkerframework.framework.flow.CFValue;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+
+/**
+ * The default transfer function for an accumulation checker.
+ *
+ * <p>Subclasses should call the {@link #accumulate(Node, TransferResult, String...)} accumulate} or
+ * {@link #accumulate(MethodInvocationNode, TransferResult, String...)} methods to accumulate a
+ * string at a particular program point.
+ */
+public class AccumulationTransfer extends CFTransfer {
+
+    protected final AccumulationAnnotatedTypeFactory typeFactory;
+
+    public AccumulationTransfer(CFAnalysis analysis) {
+        super(analysis);
+        typeFactory = (AccumulationAnnotatedTypeFactory) analysis.getTypeFactory();
+    }
+
+    /**
+     * Updates the estimate of how many things node has accumulated to include all the values.
+     *
+     * @param node the node whose estimate should be adjusted
+     * @param result the result containing the store to be modified
+     * @param values the new accumulation values
+     */
+    public void accumulate(Node node, TransferResult<CFValue, CFStore> result, String... values) {
+        AnnotatedTypeMirror oldType = typeFactory.getAnnotatedType(node.getTree());
+        AnnotationMirror newAnno = getNewAnno(oldType, values);
+        insertIntoStores(result, node, newAnno);
+    }
+
+    /**
+     * Updates the estimate for the receiver and any other receiver-returning methods in a chain
+     * with this MethodInvocationNode to include the new values.
+     *
+     * <p>For example, if the argument is the expression {@code a.b().c()}, the new value is "foo",
```

Comment:
This will only be true if methods `b` and `c` are deterministic.  If they are not, only the estimate of `a` will be updated, and the types of `a.b()` and `a.b().c()` are computed in the type factory.  We should update the documentation to be clearer about this